x-pack/plugins/actions/server/builtin_action_types/lib/send_email.ts

```diff
@@ -47,15 +59,121 @@ export interface Content {
   message: string;
 }
 
-// send an email
 export async function sendEmail(logger: Logger, options: SendEmailOptions): Promise<unknown> {
+  const { transport, content } = options;
+  const { message } = content;
+  const messageHTML = htmlFromMarkdown(logger, message);
+
+  if (transport.service === AdditionalEmailServices.EXCHANGE) {
+    return await sendEmailWithExchange(logger, options, messageHTML);
+  } else {
+    return await sendEmailWithNodemailer(logger, options, messageHTML);
+  }
+}
+
+// send an email using MS Exchange Graph API
+async function sendEmailWithExchange(
+  logger: Logger,
+  options: SendEmailOptions,
+  messageHTML: string
+): Promise<unknown> {
+  const { transport, configurationUtilities } = options;
+  const { clientId, clientSecret, tenantId, oauthTokenUrl } = transport;
+  // request access token for microsoft exchange online server with Graph API scope
+
+  const tokenResult = await requestOAuthClientCredentialsToken(
+    oauthTokenUrl ?? `${EXCHANGE_ONLINE_SERVER_HOST}/${tenantId}/oauth2/v2.0/token`,
+    logger,
+    {
+      scope: GRAPH_API_OAUTH_SCOPE,
+      clientId,
+      clientSecret,
+    },
+    configurationUtilities
+  );
+  const headers = {
+    'Content-Type': 'application/json',
+    Authorization: `${tokenResult.tokenType} ${tokenResult.accessToken}`,
+  };
+  try {
+    return await sendEmailGraphApi(
```

Comment:
Tried this out and had my permissions set incorrectly and got this error:

```
result {"error":{"code":"NoPermissionsInAccessToken","message":"The token contains no permissions, or permissions can not be understood.","innerError":{"oAuthEventOperationId":"891035f5-43d2-4915-af0b-9f761521f9b9","oAuthEventcV":"bI3EAnEB3wfuWyW0uc3koQ.1.1","errorUrl":"https://aka.ms/autherrors#error-InvalidGrant","requestId":"704d8f14-399b-4682-b89a-f4da78f6cde7","date":"2021-09-15T13:14:18"}}}
```

However the error was in the axios response and thus was not caught or logged. Can we add some error handling in `sendEmailGraphApi` to check for errors in the response?